.github/workflows/reviewdog.yml

```diff
@@ -0,0 +1,41 @@
+name: linter-checks
+on: [pull_request]
+jobs:
+  cpp:
+    permissions:
+      contents: read
+      checks: write
+      issues: write
+      pull-requests: write
+    name: pre-commit
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.11
+
+      - name: Cache pre-commit hooks
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pre-commit
+          key: ${{ runner.os }}-precommit-${{ hashFiles('.pre-commit-config.yaml') }}
+
+      - name: Install dependencies
+        run: |
+          pip install pre-commit
+
+      - name: Run pre-commit hooks
+        run: |
+          pre-commit run --all-files
+        continue-on-error: true
```

Comment:
If a stand-alone workflow is needed, you can replace these with `pre-commit/action@v3.0.1`.

Would recommend linting only on changes, see:
https://github.com/beman-project/exemplar/blob/bd7ad354842e424ca7dd69da7dff5bfaab70e8f6/.github/workflows/pre-commit.yml#L16-L24
for an implementation example.

Reply:
done